Makefile

```diff
@@ -18,16 +18,12 @@ FFT = generic# generic,fftw3,mkl
 #######CMP settings###########
 ifeq ($(CMP),intel)
 FC = mpiifort
-#FFLAGS = -fpp -O3 -xHost -heap-arrays -shared-intel -mcmodel=large -safe-cray-ptr -g -traceback
-FFLAGS = -fpp -O3 -xSSE4.2 -axAVX,CORE-AVX-I,CORE-AVX2 -ipo -fp-model fast=2 -mcmodel=large -safe-cray-ptr -I$(MPI_ROOT)/lib
-##debuggin test: -check all -check bounds -chintel eck uninit -gen-interfaces -warn interfaces
+FFLAGS = -fpp -O3 -mavx2 -mlzcnt -march=core-avx2 -mf16c
+FFLAGS += -fopenmp -I$(MPI_ROOT)/lib
 else ifeq ($(CMP),gcc)
 FC = mpif90
-#FC = mpif90-mpich-mp
-#FFLAGS = -O3 -funroll-loops -floop-optimize -g -Warray-bounds -fcray-pointer -x f95-cpp-input
-FFLAGS = -cpp -Mfree -Kieee -Minfo=accel -g -acc -target=gpu 
-#-cpp -O3 -funroll-loops -floop-optimize -g -Warray-bounds -fcray-pointer -fbacktrace -ffree-line-length-none -fallow-argument-mismatch
-#-ffpe-trap=invalid,zero
+FFLAGS = -cpp -O3 -march=native
+FFLAGS += -fopenmp -ftree-parallelize-loops=12
```

Comment:
Please note that the number of threads is specific to the test machine I am using